### PR TITLE
USEID-582: Add ADR for certificate pinning

### PR DESCRIPTION
**Pull request by mpanne**
The pen test results recommended to use certificate pinning between our app and our server. While this makes sense from a security point of view, operationally this poses a couple of challenges. The added ADR shall explain why we will not implement certificate pinning.

### CODE REVIEW

File: doc/adr/0006-no-certificat-pinning.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Decision
+
+The eID-Client will not restrict connections to servers with specific certificates since those 
+would need to be hardcoded in the eID-Client which is not possible since the servers are only
+known on start of the identification flow.
+
+## Consequences
+
+Man-in-the-middle attacks on the TLS connection are possible if an attack can get in possession of a certificate
```
Comment (ChaosCoder) [inline doc/adr/0006-no-certificat-pinning.md:27]: Suggestion (sentence was just rewritten):
```suggestion
A man-in-the-middle attack on the TLS connection is possible if an attacker comes into possession of a certificate signed by a trusted root CA for a domain used in the identification flow.
```

**Review by ChaosCoder — APPROVED** [review]
👍

File: doc/adr/0006-no-certificat-pinning.md (hunk 1 above)
Comment (mpanne) [reply doc/adr/0006-no-certificat-pinning.md]: Thanks! Suggestion applied. :)

**Review by pgurusinga — APPROVED** [review]

**Review by FabioTacke — COMMENTED** [review]
I agree. 👍

File: doc/adr/0006-no-certificat-pinning.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Date: 2022-12-01
+
+## Status
+
+Accepted
+
+## Context
+
+During the identification flow, the eID-Client connects to the server of a Service Provider (eService) and 
+to an eID-Server. Therefor, a TLS connection is being established and the eID-Client validates the server identity
```
Comment (phuesler) [inline doc/adr/0006-no-certificat-pinning.md:12]: `Therefore`

File: doc/adr/0006-no-certificat-pinning.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Date: 2022-12-01
+
+## Status
+
+Accepted
+
+## Context
+
+During the identification flow, the eID-Client connects to the server of a Service Provider (eService) and
```
Comment (phuesler) [inline doc/adr/0006-no-certificat-pinning.md:11]: Is it really called 'eService'?

File: doc/adr/0006-no-certificat-pinning.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+## Decision
+
+The eID-Client will not restrict connections to servers with specific certificates since those 
+would need to be hardcoded in the eID-Client which is not possible since the servers are only
+known on start of the identification flow.
+
+## Consequences
+
+A man-in-the-middle attack on the TLS connection is possible if an attacker comes into possession of a 
+certificate signed by a trusted root CA for a domain used in the identification flow.
```
Comment (phuesler) [inline doc/adr/0006-no-certificat-pinning.md:28]: A direct consequence is that it is easier to inspect the traffic between the eID-Client and the eService by installing and trust a Root CA on the mobile device and then use software like Charles Proxy etc. to inspect the traffic.

File: doc/adr/0006-no-certificat-pinning.md (hunk 2 above)
Comment (mpanne) [reply doc/adr/0006-no-certificat-pinning.md:12]: I used "Therefor" on purpose. Is that not how you use it? See [here](https://www.grammarly.com/blog/therefore-vs-therefor/#:~:text=Therefore%20is%20an%20adverb%20that,%E2%80%9D%20or%20%E2%80%9Cfor%20it.%E2%80%9D)

File: doc/adr/0006-no-certificat-pinning.md (hunk 3 above)
Comment (mpanne) [reply doc/adr/0006-no-certificat-pinning.md:11]: "eService" is the term for Service Provider (Diensteanbieter) which is used in the technical guidelines.

File: doc/adr/0006-no-certificat-pinning.md (hunk 2 above)
Comment (phuesler) [reply doc/adr/0006-no-certificat-pinning.md:12]: Oh, I did not know that word existed.

File: doc/adr/0006-no-certificat-pinning.md (hunk 4 above)
Comment (phuesler) [reply doc/adr/0006-no-certificat-pinning.md:28]: certificate pinning can also make it more difficult to debug

**Pull request merged**